includes/class-windows-azure-helper.php

```diff
@@ -579,11 +578,53 @@ static public function file_exists( $relative_path ) {
 		$exist = false;
 
 		if ( WP_Filesystem() ) {
-			$upload_dir = wp_upload_dir();
-			$filename   = trailingslashit( $upload_dir['basedir'] ) . $relative_path;
+			$filename   = trailingslashit( self::get_uploads_basedir() ) . $relative_path;
 			$exist = $wp_filesystem->exists( $filename, false, 'f' );
 		}
 
 		return apply_filters( 'windows_azure_storage_file_exist', $exist, $relative_path );
 	}
+
+    /**
+     * Get the uploads basedir without `sites/SITE_ID` for multisite installation.
+     *
+     * @since 4.2.0
+     *
+     * @return bool
+     */
+    static public function get_uploads_basedir() {
+        $upload_dir = wp_upload_dir();
+        return preg_replace( '#\/sites\/[0-9]+#', '', $upload_dir['basedir'] );
```

Comment:
@ravichandrach1, why do you remove `sites/{id}` from base dir? This is the root of the problem, isn't it? We have to respect sites folders to make sure we don't have image conflicts.

Reply:
@eugene-manuilov The problem was the plugin was removing the `basedir` part of the path and only adding the `subdir` path to the blob name. `wp_upload_dir` returns the `sites/{id}` part in the `basedir` and not in `subdir`. These 2 functions i wrote will remove the `sites/{id}` part from the `basedir` and append it to the `subdir` so that the blob name will contain the full blob name including the sites part.